### ISSUE TITLE
[v2 FEATURE]: chemprop_train: take advantage of multiple CPUs

### ISSUE THREAD

**Issue by UnixJunkie** [chemprop/chemprop#847]
**Is your feature request related to a problem? Please describe.**

chemprop_train does not seem to take advantage of parallel computers

**Use-cases/examples of this new feature**

train models faster

**Desired solution/workflow**

accept a -nprocs or -np CLI option; or exploit the value in OMP_NUM_THREADS

**Discussion**

We are in 2024 gentlemen... supercomputer nodes might have 56 CPUs but no gfx card...

**Additional context**

Datasets in 2024 might also have 100k to millions of molecules...

**Comment by JacksonBurns** [comment]
> chemprop_train does not seem to take advantage of parallel computers

It does.
Chemprop v1 allows passing `--num_workers` to control how many processes are used for data loading, as stated in the README (see https://github.com/chemprop/chemprop/tree/master?tab=readme-ov-file#caching).

Chemprop v2 supports this and adds support for training in parallel using Lightning's DDP, which will not be backported to v1. I'll note that torch already does a lot of multiprocessing in the background when running on CPU though.

> accept a -nprocs or -np CLI option; or exploit the value in OMP_NUM_THREADS

See above for v1, see the new docs page for v2 parallel data loading: https://chemprop.readthedocs.io/en/main/cmd.html#dataloader-args.

> We are in 2024 gentlemen... supercomputer nodes might have 56 CPUs but no gfx card...

Our team has both men and women, please use inclusive language in the future.
